server/routes/user/index.ts

```diff
@@ -534,26 +533,17 @@ router.get<{ id: string }, UserWatchDataResponse>(
 
       const media = (
         await Promise.all(
-          uniqWith(watchHistory, (recordA, recordB) =>
-            recordA.grandparent_rating_key && recordB.grandparent_rating_key
-              ? recordA.grandparent_rating_key ===
-                recordB.grandparent_rating_key
-              : recordA.parent_rating_key && recordB.parent_rating_key
-              ? recordA.parent_rating_key === recordB.parent_rating_key
-              : recordA.rating_key === recordB.rating_key
+          watchHistory.map(
+            async (record) =>
+              await mediaRepository.findOne({
+                where: {
+                  ratingKey:
+                    record.media_type === 'movie'
+                      ? record.rating_key
+                      : record.grandparent_rating_key,
+                },
+              })
```

Comment:
Just realized how this is being done. We should refactor this as well to use an IN so we can query all the media at once?

Reply:
Oh yup, we can do that now!  Will fix.